src/pages/matching/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { api } from "../../utils/api";
+import DefaultLoadingPage from "../../components/loading/loading";
+import { useSession } from "next-auth/react";
+import { useState, useMemo } from "react";
+import { Alert, AlertTitle } from "@mui/material";
+import MatchingFormSteps from "../../components/matching/MatchingFormSteps";
+import PleaseSignInPage from "../../components/PleaseSignInPage";
+import { type Preference } from "../../components/matching/MatchingFormSteps";
+import { type RouterOutputs } from "../../utils/api";
+
+export type getByRoleOutput = RouterOutputs["user"]["getByRole"];
+export type getByRoleOutputData = getByRoleOutput[0];
+
+const formatOptionsObject = (match: getByRoleOutputData): Preference => {
+  const firstLetter = match?.user.name ? match?.user.name.charAt(0) : "N";
+  return {
+    name: match?.user.name ? match?.user.name : "No Name",
+    id: match?.user?.id,
+    firstLetter: /[a-zA-Z]/.test(firstLetter) ? firstLetter : "#",
+  };
+};
+
+const MatchingPage = () => {
+  const { data: sessionData } = useSession();
+  const [submitted, setSubmitted] = useState(false);
+
+  const otherRole =
+    sessionData?.user?.roleName === "MENTEE" ? "MENTOR" : "MENTEE";
+
+  const onSubmit = (preferences: Preference[]) => {
+    // TODO:
+    // Add your form submission logic here (e.g., API call)
+    // mutation.mutate(result.data);
+    console.log("Final preferences order:", preferences);
+    setSubmitted(true);
+  };
+
+  // if mentee, query all mentors and vice versa
+  const {
+    data: allMatches,
+    error,
+    isLoading,
+  } = api.user.getByRole.useQuery({ role: otherRole });
+
+  const preferenceOptions = useMemo(() => {
+    let options: Preference[] = [];
+    if (!allMatches) return options;
+
+    options = allMatches
+      .map(formatOptionsObject)
+      .toSorted((a, b) => -b.firstLetter.localeCompare(a.firstLetter));
```

Comment:
Haha, `-b.firstLetter.localeCompare(a.firstLetter)` = `a.firstLetter.localeCompare(b.firstLetter)`

Reply:
Same thing no need to change

Reply:
taken from MUI docs lo, can change it tho.